### PR TITLE
Spring boot 3.3 and Java 17 upgrade

### PR DESCRIPTION
**Pull request by patope**

### CODE REVIEW

File: dialob-build-parent/pom.xml [hunk 1]
```diff
@@ -30,87 +30,84 @@
...
+    <hibernate-validator.version>8.0.1.Final</hibernate-validator.version>
+    <immutables.version>2.10.1</immutables.version>
+    <jakarta.servlet-api.version>6.0.0</jakarta.servlet-api.version>
+    <json-path.version>2.9.0</json-path.version>
     <jsoup.version>1.15.3</jsoup.version>
     <feign.version>11.8</feign.version>
 
-    <junit.jupiter.version>5.8.2</junit.jupiter.version>
-    <junit.platform.version>1.8.2</junit.platform.version>
+    <junit.jupiter.version>5.10.2	</junit.jupiter.version>
```
Comment (olev-mutso) [inline dialob-build-parent/pom.xml:69]: Remove spaces after version number.

File: docs/spring-boot-3.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+### Changed properties
+
+|| Old name         || New name              ||
```
Comment (villuv) [inline docs/spring-boot-3.md:4]: Table formatting is broken

**Review by villuv — CHANGES_REQUESTED** [review]

**Comment by villuv** [comment]
dialob-boot and dialob-session-boot Dockerfiles need to switch to java 17 base image

**Review by olev-mutso — CHANGES_REQUESTED** [review]
Have you checked findbugs report?

**Review by villuv — CHANGES_REQUESTED** [review]
I built local docker images from this branch, replaced an existing local dev environment setup using v. 2.1 along with changing the redis parameter as instructed. Local configuration has ` DIALOB_SECURITY_ENABLED=false` configured.

I get 403 on OPTIONS http://localhost:8085/dialob/api/forms/xxx request

So either migration instructions are not complete or security disabled mode doesn't work.

File: dialob-components/dialob-questionnaire-service/src/main/java/io/dialob/questionnaire/service/DialobQuestionnaireServiceAutoConfiguration.java [hunk 3]
```diff
@@ -74,7 +73,7 @@ public PostSubmitHandler postSubmitHandler() {
   }
 
   @Bean
-  public EventPublisher questionnaireSessionEventPublisher(TaskExecutor taskExecutor, ApplicationEventPublisher applicationEventPublisher) {
+  public EventPublisher questionnaireSessionEventPublisher(@Qualifier("applicationTaskExecutor") TaskExecutor taskExecutor, ApplicationEventPublisher applicationEventPublisher) {
```
[the remark this answers is not in the document]
Comment (patope) [reply dialob-components/dialob-questionnaire-service/src/main/java/io/dialob/questionnaire/service/DialobQuestionnaireServiceAutoConfiguration.java:76]: Name is `applicationTaskExecutor` provided by Spring's `@EnableTaskExecutor`. `@EnableTaskScheduling` and `@EnableTaskExecutor` both define executors and trigger injection conflict.

File: dialob-components/dialob-integration-api/src/main/java/io/dialob/integration/api/event/FormTaggedEvent.java [hunk 4]
```diff
@@ -18,22 +18,21 @@
...
 
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
-
 
 @Value.Immutable
 @JsonSerialize(as = ImmutableFormTaggedEvent.class)
 @JsonDeserialize(as = ImmutableFormTaggedEvent.class)
 @JsonInclude(content = JsonInclude.Include.NON_NULL, value = JsonInclude.Include.NON_EMPTY)
 public interface FormTaggedEvent extends FormEvent, DistributedEvent {
```
[the remark this answers is not in the document]
Comment (patope) [reply dialob-components/dialob-integration-api/src/main/java/io/dialob/integration/api/event/FormTaggedEvent.java:31]: It's not for `@Value.Immutable`. It's for semantic analyzers.

**Comment by patope** [comment]
> **villuv ** requested changes

Auto configurations didnt work. Try again.

File: dialob-components/dialob-questionnaire-service/src/main/java/io/dialob/questionnaire/service/DialobQuestionnaireServiceAutoConfiguration.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (patope) [reply dialob-components/dialob-questionnaire-service/src/main/java/io/dialob/questionnaire/service/DialobQuestionnaireServiceAutoConfiguration.java]: Replaced with constant `TaskExecutionAutoConfiguration.APPLICATION_TASK_EXECUTOR_BEAN_NAME`

File: dialob-components/dialob-cache/src/main/java/io/dialob/cache/DialobCacheAutoConfiguration.java [hunk 5]
```diff
@@ -63,7 +63,7 @@ public CacheManager sessionCacheManager() {
     }
 
     @Bean
-    public ScheduledSessionEvictionPolicy scheduledSessionEvictionPolicy(Clock clock, Optional<QuestionnaireSessionSaveService> sessionService, Optional<CacheManager> cacheManager, DialobSettings dialobSettings) {
+    public ScheduledSessionEvictionPolicy scheduledSessionEvictionPolicy(Optional<Clock> clock, Optional<QuestionnaireSessionSaveService> sessionService, Optional<CacheManager> cacheManager, DialobSettings dialobSettings) {
```
[the remark this answers is not in the document]
Comment (patope) [reply dialob-components/dialob-cache/src/main/java/io/dialob/cache/DialobCacheAutoConfiguration.java]: Remove Clock bean

File: dialob-session-boot/src/main/resources/application.yml [hunk 6]
```diff
@@ -97,7 +97,7 @@ dialob:
     sockjs:
       enabled: true
       webSocketEnabled: true
```
[the remark this answers is not in the document]
Comment (patope) [reply dialob-session-boot/src/main/resources/application.yml:99]: https://github.com/dialob/dialob-parent/blob/62a04f59b300570130fe640b4b3a58bcde79c133/docs/spring-boot-3.md

File: dialob-components/dialob-session-engine/src/main/java/io/dialob/session/engine/program/model/DisplayItem.java [hunk 7]
```diff
@@ -15,9 +15,9 @@
  */
 package io.dialob.session.engine.program.model;
 
+import edu.umd.cs.findbugs.annotations.NonNull;
```
[the remark this answers is not in the document]
Comment (patope) [reply dialob-components/dialob-session-engine/src/main/java/io/dialob/session/engine/program/model/DisplayItem.java:18]: Uses only `edu.umd.cs.findbugs.annotations.NonNull` now

**Pull request merged**